docs/guides/server/logging.adoc

```diff
@@ -71,15 +74,90 @@ To enable log handlers, enter the following command:
+
+It means properties for log level settings look like this:
+
+* `log-console-level` - Console log handler
+* `log-file-level` - File log handler
+* `log-syslog-level` - Syslog log handler
+
+NOTE: The `log-<handler>-level` properties are available only when the particular log handlers are enabled.
+More information in log handlers settings below.
+
+Accepted log levels are specified in <<Log levels>> section, and *must be in lowercase*.
+
+=== General principle
+
+It is necessary to understand that setting the log levels for each particular handler *does not override the root level* specified in the `log-level` property.
+It specifies the minimal allowed level for logs shown in the output.
+It respects levels specified in the `log-level` property, which creates a minimal log level bar for log records.
+
+Specifically, when an arbitrary log level is specified for the handler, it does not mean the log records with the log level will be present in the output.
+The root log level needs to be assessed as well, as it determines the minimal level for exposed log records.
+
+Log handler levels provide the *restriction for the root log level*, so the default value is `all` - without any restriction.
+
+==== Visualization
+
+To better understand the principle of logging levels, the visualization of log level settings might help.
+Only the `log-level` property is changed, and other log handlers are set as follows:
+
+<@kc.start parameters="--log=syslog,console,file --log-level=info --log-syslog-level=all --log-console-level=debug --log-file-level=trace"/>
+
+.Root log level `info`
+[#img-log-level-info]
+image::logging/log-levels-root-info.png[Root log level info,720]
+
+As you can see in the figure <<img-log-level-info>>, the root log level, specified via the `log-level` property, sets the minimal level for log records.
+Therefore, even when their log level is the finest, it is ignored and respects the parent level.
+
+The _Min Level_ bar represents the minimal allowed log level in the application (`trace` by default).
+Log records with finer log levels will never be present in {project_name}.
+It is controlled by the Quarkus property `quarkus.log.min-level`.
+
+.Root log level `trace`
+[#img-log-level-trace]
+image::logging/log-levels-root-trace.png[Root log level trace,720]
+
+In this case, by changing the `log-level` property to `trace` level, visible in the figure <<img-log-level-trace>>, log handlers are not restricted anymore.
+
+.Additional categories
+[#img-log-level-categories]
+image::logging/log-levels-root-info-categories.png[Additional categories,720]
+
+As mentioned in the <<Configuring category-specific log levels>> section, some additional categories can be set.
+In figure <<img-log-level-categories>>, the `org.keycloak.events` logger uses the `debug` log level, where its log records will be part of the output next to the `info` log records.
```

Comment:
This feels a bit confusing and IMHO is making the impression we support setting categories on log handler level. I'd rephrase the whole paragraph a bit.